trace-utils/src/msgpack_decoder/v04/decoder/span.rs

```diff
@@ -26,103 +26,77 @@ use std::str::FromStr;
 /// This function will return an error if:
 /// - The map length cannot be read.
 /// - Any key or value cannot be decoded.
-#[inline]
-pub(crate) fn decode_span(buf: &mut &[u8]) -> Result<Span, DecodeError> {
+pub fn decode_span<'a>(
+    buffer: &'a tinybytes::Bytes,
```

Comment:
Can't we replace `buf` with `buffer`completely ? And have all the function decoding msgpack return slice of `buffer` which would also be Bytes.

Reply:
The `rmp` library code expects &mut [u8], so we need to either pass it around or do a conversion every time from tinybytes to slice. I can include the underlying &mut [u8] in the `BufferWrapper` struct to make it more ergonomic.